### PR TITLE
Registration dev

### PR DESCRIPTION
**Pull request by basvwijk**

### CODE REVIEW

File: app/controllers/UserController.java [hunk 1]
```diff
@@ -36,16 +36,36 @@ public Result createUser() {
...
         user.setUsername(body.get("username").asText());
-        user.setPassword(Crypt.crypt(body.get("password").asText()));
-        userRepository.createUser(user);
-        try {
-            return created(mapper.writeValueAsString(user));
-        } catch (JsonProcessingException e) {
-            e.printStackTrace();
-            return internalServerError(e.getMessage());
+        if(!userRepository.checkIfExists(user.getUsername())){
+            user.setPassword(Crypt.crypt(body.get("password").asText(), "LeagueofLegends"));
```
Comment (C-Alexander) [inline app/controllers/UserController.java]: pls remove enter

File: app/controllers/UserController.java [hunk 2]
```diff
@@ -36,16 +36,36 @@ public Result createUser() {
...
+            userRepository.createUser(user);
+            try {
+                return created(mapper.writeValueAsString(user));
+            } catch (JsonProcessingException e) {
+                e.printStackTrace();
```
Comment (C-Alexander) [inline app/controllers/UserController.java]: Use the logger. Logger.error() is the correct one.

File: app/controllers/UserController.java [hunk 3]
```diff
@@ -36,16 +36,36 @@ public Result createUser() {
...
+                return internalServerError(e.getMessage());
+            }
+        }
+        else {
+            return status(409, "Username already exists, have you tried logging in?");
         }
     }
 
+    @Transactional
+    public Result loginUser() {
```
Comment (C-Alexander) [inline app/controllers/UserController.java]: User isn't strictly needed in terms of naming. Is there going to be a loginDog?

File: app/controllers/UserController.java [hunk 4]
```diff
@@ -36,16 +36,36 @@ public Result createUser() {
...
+        JsonNode body = request().body().asJson();
+        User user = new User();
+        UUID uuid = UUID.randomUUID();
+        user.setUsername(body.get("username").asText());
+        user.setPassword(Crypt.crypt(body.get("password").asText(), "LeagueofLegends"));
+        System.out.println(user.getPassword());
```
Comment (C-Alexander) [inline app/controllers/UserController.java]: Don't system.out, use Logger.info

File: app/dal/entities/User.java [hunk 5]
```diff
@@ -6,7 +6,9 @@
 @Table(name = "Player")
 @NamedQueries({
         @NamedQuery(name = "User.getAll", query = "select u from User as u"),
-        @NamedQuery(name = "User.findOne", query = "select u from User as u WHERE u.id = :id")
+        @NamedQuery(name = "User.findOne", query = "select u from User as u WHERE u.id = :id"),
+        @NamedQuery(name = "User.ifExists", query = "select (COUNT(*) > 0) as exists  from User as u WHERE u.username = :username"),
+        @NamedQuery(name = "User.login", query = "select (COUNT(*) > 0) as exists  from User as u WHERE u.username = :username and u.password = :password")
```
Comment (C-Alexander) [inline app/dal/entities/User.java]: Could also make it so you give it a UUID, if it finds one it updates it and returns it, if it doesnt it returns 0 and you say as much

File: app/dal/repositories/UserRepository.java [hunk 6]
```diff
@@ -27,4 +24,12 @@ public void createUser(User user) {
     public List<User> getAllUsers() {
         return context.findAll();
     }
+
+    public Boolean checkIfExists(String username) {
```
Comment (C-Alexander) [inline app/dal/repositories/UserRepository.java]: maybe change this to simply ifExists

File: app/dal/entities/User.java (hunk 5 above)
Comment (joeyboy108) [reply app/dal/entities/User.java]: I dont get what you mean. Could you explain yourself?

**Pull request merged**